src/utils/screenshotsaver.cpp

```diff
@@ -99,6 +99,55 @@ bool ScreenshotSaver::saveToFilesystem(const QPixmap& capture,
     return ok;
 }
 
+QString ScreenshotSaver::ShowSaveFileDialog(QWidget* parent,
+                                            const QString& title,
+                                            const QString& directory,
+                                            const QString& filter)
+{
+#if defined(Q_WS_WIN) || defined(Q_WS_MAC)
+    return QFileDialog::getSaveFileName(parent, title, directory, filter);
+#else
+    QFileDialog dialog(parent, title, directory, filter);
+    if (parent) {
+        dialog.setWindowModality(Qt::WindowModal);
+    }
+
+    QRegExp filter_regex(QLatin1String("(?:^\\*\\.(?!.*\\()|\\(\\*\\.)(\\w+)"));
```

Comment:
Could you define this regex as a variable name, move to a function, or add a comment to explain what it does? I'm not a regex wizard :)

Reply:
I changed the regex to some QString operations (extracting extension from chosen filter). IMHO is easier to understand it now.